modules/core/src/main/java/org/apache/synapse/message/store/impl/rabbitmq/RabbitMQProducer.java

```diff
@@ -36,147 +36,186 @@
-		try {
-			//Serializing message
-			ByteArrayOutputStream os = new ByteArrayOutputStream();
-			ObjectOutput objOut = new ObjectOutputStream(os);
-			objOut.writeObject(message);
-			byte[] byteForm = os.toByteArray();
-			objOut.close();
-			os.close();
-			//building AMQP message
-			AMQP.BasicProperties.Builder builder = new AMQP.BasicProperties().builder();
-			builder.messageId(synCtx.getMessageID());
-			builder.deliveryMode(MessageProperties.MINIMAL_PERSISTENT_BASIC.getDeliveryMode());
-			builder.priority(message.getPriority(DEFAULT_PRIORITY));
-			channel = connection.createChannel();
-			if (exchangeName == null) {
-				channel.basicPublish("", queueName, builder.build(), byteForm);
-			} else {
-				channel.basicPublish(exchangeName, queueName, builder.build(), byteForm);
-			}
-		} catch (IOException e) {
-			throwable = e;
-			error = true;
-			isConnectionError = true;
-		} catch (Throwable t) {
-			throwable = t;
-			error = true;
-		} finally {
-			if (channel != null && channel.isOpen())
-				try {
-					channel.close();
-				} catch (TimeoutException e) {
-					logger.error("Error when closing connection,TimeoutException" + synCtx.getMessageID() + ". " , e);
-				} catch (IOException e) {
-					logger.error(
-							"Error when closing connection" + synCtx.getMessageID() + ". " + e);
-				}
-		}
-		if (error) {
-			String errorMsg = getId() + ". Ignored MessageID : " + synCtx.getMessageID()
-			                  + ". Could not store message to store ["
-			                  + store.getName() + "]. Error:" + throwable.getLocalizedMessage();
-			logger.error(errorMsg, throwable);
-			store.closeProducerConnection();
-			connection = null;
-			if (logger.isDebugEnabled()) {
-				logger.debug(getId() + ". Ignored MessageID : " + synCtx.getMessageID());
-			}
-			return false;
-		}
-		if (logger.isDebugEnabled()) {
-			logger.debug(getId() + ". Stored MessageID : " + synCtx.getMessageID());
-		}
-		store.enqueued();
-		return true;
-	}
-
-	/**
-	 * Used to close the channel opened in this object instance.
-	 * This should be called after the end of each call on storeMessage method
-	 * But instead of this, finally block is used in storeMethod to close the channels
-	 */
-	public boolean cleanup() {
-		return store.cleanup(null, false);
-	}
-
-	public boolean isInitialized() {
-		return isInitialized;
-	}
-
-	public String getId() {
-		return idString;
-	}
+    private static final Log log = LogFactory.getLog(RabbitMQProducer.class.getName());
+
+    private static final int DEFAULT_PRIORITY = 0;
+    private Connection connection;
+    private RabbitMQStore store;
+    private String routingKey;
+    private String exchangeName;
+    private boolean isInitialized = false;
+    private String idString; // ID of the MessageProducer
+
+    /**
+     * The RabbitMQ producer
+     *
+     * @param store the {@link RabbitMQStore} object
+     */
+    public RabbitMQProducer(RabbitMQStore store) {
+        if (store == null) {
```

Comment:
Same as consumer. If parameters are null shall we throw exceptions so that deployment fails with required message?

Reply:
That is how it was handled in other types of store processor as well. Technically, the store won't become null if you deploy the processor by creating through management console or tooling.